### PR TITLE
Double-check pod readiness for deployments

### PR DESCRIPTION
**Pull request by unguiculus**
I took a stab at #9. This is still a WIP. I need to do more testing.

### CODE REVIEW

**Comment by unguiculus** [comment]
cc @prydonius @mattfarina

**Comment by prydonius** [comment]
This looks like a better solution than my suggestion of grepping, nice @unguiculus 😃!

File: lib/chartlib.sh [hunk 1]
```diff
@@ -289,11 +289,30 @@ chartlib::install_chart_with_single_config() {
...
+            # 'kubectl rollout status' does not return a non-zero exit code when rollouts fail.
+            # We, thus, need to double-check here.
+
+            local jsonpath='{.status.conditions[?(@.type=="Ready")].status}'
+
+            for pod in $(chartlib::get_pods_for_deployment "$deployment" "$namespace"); do
+                ready=$(kubectl get pod "$pod" --namespace "$namespace" --output jsonpath="$jsonpath")
+                if [[ "$ready" != "True" ]]; then
+                    chartlib::error "Pod '$pod' did not reach ready state!"
+                    error=true
```
Comment (prydonius) [inline lib/chartlib.sh]: why not just `return 1` here, or is your idea to enumerate all the pods in all the deployments that don't go to running?

Comment (unguiculus) [reply lib/chartlib.sh]: Yes, the idea was to enumerate the pods.

**Comment by unguiculus** [comment]
I tested it with https://github.com/helm/charts/pull/2766. Looks good now.

```
Installing chart 'stable/portus' into namespace 'portus-iz51b4b5ici6mc97'...
Chart does not provide test values. Using defaults...
Error: release portus-iz51b4b5ici6mc97 failed: timed out waiting for the condition
Waiting for rollout to finish: 0 of 1 updated replicas are available...
error: deployment "portus-iz51b4b5ici6mc97" exceeded its progress deadline
ERROR: Pod 'portus-iz51b4b5ici6mc97-754d8bb44-pthrg' did not reach ready state!
error: deployment "portus-iz51b4b5ici6mc97-docker-registry" exceeded its progress deadline
ERROR: Pod 'portus-iz51b4b5ici6mc97-docker-registry-74d44cd9b-h5jmn' did not reach ready state!
ERROR: Pod 'portus-iz51b4b5ici6mc97-docker-registry-74d44cd9b-vwj2j' did not reach ready state!
deployment "portus-iz51b4b5ici6mc97-mariadb" successfully rolled out
```

**Comment by unguiculus** [comment]
cc @helm/charts-maintainers

File: lib/chartlib.sh (hunk 1 above)
Comment (scottrigby) [reply lib/chartlib.sh]: I like this much better than grepping output for `error` 👍

File: README.md [hunk 2]
```diff
@@ -72,7 +72,7 @@ chart-test.sh
 ### Linting Charts
 
 ```shell
-docker run --rm -v "$(pwd):/workdir" --workdir /workdir gcr.io/kubernetes-charts-ci/chart-testing:v1.0.2 chart_test.sh --no-install --config .mytestenv
+docker run --rm -v "$(pwd):/workdir" --workdir /workdir gcr.io/kubernetes-charts-ci/chart-testing:v1.0.3 chart_test.sh --no-install --config .mytestenv
```
Comment (scottrigby) [inline README.md]: I feel like I should know this, but what's our process for tagging/pushing this image?

Comment (unguiculus) [reply README.md]: There's not really a process. So far I've been pushing the images manually which is not ideal. See #4.

**Comment by unguiculus** [comment]
Pushed image and created https://github.com/helm/charts/pull/7203.

**Pull request merged**